### PR TITLE
[Backup] Added multiple files restore capabilities

### PR DESCRIPTION
**Pull request by sambitratha**
**History Notes:**  
(Fill in the following template if multiple notes are needed, otherwise PR title will be used for history note.)

[Component Name 1] (BREAKING CHANGE:) (az command:) make some customer-facing change.  
[Component Name 2] (BREAKING CHANGE:) (az command:) make some customer-facing change.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [X] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [X] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
add to S167.

File: src/azure-cli/azure/cli/command_modules/backup/_params.py [hunk 1]
```diff
@@ -259,7 +259,7 @@ def load_arguments(self, _):
         c.argument('target_folder', options_list=['--target-folder'], help='Destination folder to which content will be restored. To restore content to root , leave the folder name empty')
         c.argument('target_storage_account', options_list=['--target-storage-account'], help='Destination storage account to which content will be restored')
         c.argument('source_file_type', arg_type=get_enum_type(['File', 'Directory']), options_list=['--source-file-type'], help='Specify the source file type to be selected')
-        c.argument('source_file_path', options_list=['--source-file-path'], help="""The absolute path of the file, to be restored within the file share, as a string. This path is the same path used in the 'az storage file download' or 'az storage file show' CLI commands.""")
```
Comment (qianwens) [inline src/azure-cli/azure/cli/command_modules/backup/_params.py]: >source_file_path [](start = 20, length = 16)

please add sample in help.py

Comment (qianwens) [reply src/azure-cli/azure/cli/command_modules/backup/_params.py]: please also add test case for multi file support

---
In reply to: [390721217](https://github.com/Azure/azure-cli/pull/12522#discussion_r390721217) [](ancestors = 390721217)

File: src/azure-cli/azure/cli/command_modules/backup/_params.py [hunk 2]
```diff
@@ -259,7 +259,7 @@ def load_arguments(self, _):
         c.argument('target_folder', options_list=['--target-folder'], help='Destination folder to which content will be restored. To restore content to root , leave the folder name empty')
         c.argument('target_storage_account', options_list=['--target-storage-account'], help='Destination storage account to which content will be restored')
         c.argument('source_file_type', arg_type=get_enum_type(['File', 'Directory']), options_list=['--source-file-type'], help='Specify the source file type to be selected')
```
Comment (qianwens) [inline src/azure-cli/azure/cli/command_modules/backup/_params.py]: >source_file_type [](start = 20, length = 16)

all the file must be the same type? or you should implement command like --source-file type='json' path="/e"  --source-file type='txt' path="/b"

Comment (qianwens) [reply src/azure-cli/azure/cli/command_modules/backup/_params.py]: also the target_folder

---
In reply to: [390722472](https://github.com/Azure/azure-cli/pull/12522#discussion_r390722472) [](ancestors = 390722472)

Comment (sambitratha) [reply src/azure-cli/azure/cli/command_modules/backup/_params.py]: Yes, we are allowing restore of multiple items of same type i.e. file or directory. Point to be mentioned here is source_file_type is not the extension of the file. It simply specifies if the path given by the user is a file or a directory

File: src/azure-cli/azure/cli/command_modules/backup/_params.py (hunk 1 above)
Comment (sambitratha) [reply src/azure-cli/azure/cli/command_modules/backup/_params.py]: Added

**Pull request merged**